boards/arm/sam4e_xpro/sam4e_xpro.dts

```diff
@@ -41,6 +41,10 @@
 
 	gpio_keys {
 		compatible = "gpio-keys";
+
+		pinctrl-0 = <&gpio_keys_default>;
+		pinctrl-names = "default";
```

Comment:
if you look at Linux, the way `gpio-keys` are configured as wake-up sources is via the `wakeup-source` property (in each subnode). So, I guess the proper way to handle this is to make the input driver handle such property and forward it to the GPIO/pinctrl(?) driver when configuring the pin.

Reply:
ref https://www.kernel.org/doc/Documentation/devicetree/bindings/input/gpio-keys.txt

Reply:
> if you look at Linux, the way gpio-keys are configured as wake-up sources is via the wakeup-source property (in each subnode). So, I guess the proper way to handle this is to make the input driver handle such property and forward it to the GPIO/pinctrl(?) driver when configuring the pin.

This PR seems to be the most far I can go with what is implemented in tree. IMHO the use of `wakeup-source` can be addressed in another moment because it seems to require a dedicated API which, for the moment, is out of scope for this PR. To propose an API change or new API require in deep knowledge from other architectures which I don't have.

Reply:
I'm fine adding these pinctrl entries (thanks for the https://github.com/torvalds/linux/blob/750b95887e567848ac2c851dae47922cac6db946/arch/arm/boot/dts/microchip/at91-sama5d2_xplained.dts#L708-L720 example). However, these entries should be of course optional, and the driver needs to be prepared to handle the case when they're not present, that is, handle the `-ENOENT` error case gracefully. Also, the `wakeup-source` can be used as a _toggle_ to apply such configuration, I guess.

Reply:
> I'm fine adding these pinctrl entries

Ok, thank you. Can I get your approval on #64213 since it a dependency?

> However, these entries should be of course optional, and the driver needs to be prepared to handle the case when they're not present, that is, handle the -ENOENT error case gracefully.

Perfect! I'll do it.

> Also, the wakeup-source can be used as a toggle to apply such configuration, I guess.

This is what I want add later on another PR but that is the final idea.